src/components/WalletModal/index.tsx

```diff
@@ -274,21 +280,75 @@ export default function WalletModal({
         })
     }
 
+    const handleEthereumNetworkSwitch = () => {
```

Comment:
use useCallback correctly in ALL callback functions
@adaptivecodeio this is a basic thing that should come up in your review

Reply:
noted